### ISSUE TITLE
Rule Change: Have `no-unreachable-loop` ignore async iterators

### ISSUE THREAD

**Issue by voxpelli** [eslint/eslint#15219]
### What rule do you want to change?

no-unreachable-loop

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
async function peek () {
  for await (const data of createIterator()) {
    return data
  }
  throw new Error('Not found')
}
```

### What does the rule currently do for this code?

It complains that the loop can never reach the second iteration

### What will the rule do after it's changed?

It acknowledges that using a `for await` like that is the easiest way of getting the first item from and async iterator and thus does not complain about the fact that the loop can never reach the second iteration

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

Brought up by @mafintosh in https://github.com/standard/standard/issues/1723

**Comment by ljharb** [comment]
Why not `await createIterator().next()).value`?

There's also a proposed `Array.fromAsync` so you could do `const [data] = await Array.fromAsync(createIterator());`

**Comment by nzakas** [comment]
Closing due to age
